src/lwt.erl

```diff
@@ -120,25 +225,51 @@ handle_call(
         RewardShares
     ),
     RewardShare = LWT / TotalShares,
-    NewHolders = maps:fold(
+    NewHolders0 = maps:fold(
         fun(K, V, Acc) ->
             credit(K, trunc(V * RewardShare), Acc)
         end,
         State#state.holders,
         RewardShares
     ),
-    lager:info("New Holders: ~p", [NewHolders]),
+    lager:debug("New Holders: ~p", [NewHolders0]),
+
+    UnstakedValidators = maps:keys(ChainValidators) -- maps:keys(State#state.validators),
+    lager:info("UnstakedValidators: ~p", [UnstakedValidators]),
+
+    NewHolders = lists:foldl(
+        fun(ValidatorAddress, HAcc) ->
+            {Owner, InitHeight} = maps:get(ValidatorAddress, ChainValidators),
+            case (InitHeight + ?ValidatorStakingPeriod) > ChainHt of
+                false ->
+                    HAcc;
+                true ->
+                    lager:info("Crediting owner: ~p for unstaking: ~p", [Owner, ValidatorAddress]),
+                    credit(Owner, ?ValidatorCost, HAcc)
+            end
+        end,
+        NewHolders0,
+        UnstakedValidators
+    ),
+
     %% Now we need to remove the first `OpCount' operations from our pending operations stack, zero out our burns
     %% and increment our nonce
     {reply, ok, State#state{
         nonce = Nonce + 1,
         pending_operations = lists:sublist(Ops, OpCount + 1, length(Ops)),
         burns = 0,
-        holders = NewHolders
+        holders = NewHolders,
+        chain_ht = ChainHt
     }}.
 
 debit(Key, Amount, Map) ->
     maps:update_with(Key, fun(V) -> V - Amount end, Map).
 
 credit(Key, Amount, Map) ->
     maps:update_with(Key, fun(V) -> V + Amount end, Amount, Map).
+
+add_validator(Key, Val, Map) ->
+    maps:put(Key, Val, Map).
+
+remove_validator(Key, Map) ->
```

Comment:
isn't this the point where you'd want to return whatever stake?

Reply:
oh there will be a "cooldown" period here too then? i would suggest putting the cooldown block in the stake metadata then, or maybe track cooldowns in their own datastructure by block height, then when the block height tick executes, you can check that map and see if there's any pending operations

Reply:
if we don't have some sort of periodic tick, we could force the unstaker to wait through the cooldown then issue a finalize txn that's only valid after the cooldown.